### ISSUE TITLE
Wrong type on `OlSourceVector.features`

### ISSUE THREAD

**Issue by azygis** [MelihAltintas/vue3-openlayers#399]
**Describe the bug**
The type on `features` prop of `OlSourceVector` is wrong (or incomplete). Currently it accepts array / collection of `Geometry` but it should be `Feature` or even `FeatureLike`. This forces us to add nasty hacks like `const realFeatures = computed(() => features.value as unknown as Geometry[]);` and passing that instead.

**Affected version(s)**

Pretty sure all version up until now.

**To Reproduce**
N/A

**Expected behavior**
Above.

**Screenshots**
N/A

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
I can create a pull request with the change but I'm not sure if it will become a breaking change or not. Would like to know if it was an oversight or an intentional implementation.

**Comment by d-koppenhagen** [comment]
Hey, please send me a PR. I will check it and think about if a breaking change is necessary.

**Comment by caiohamamura** [comment]
That also makes impossible to know exactly which feature it actually is when intercepting looping through `forEachFeatureAtPixel`, because the features created from the `geometry` are not the same feature as those in `features` array. So I can't do `features.indexOf(feature)`, because they are not the same object.

**Comment by azygis** [comment]
I'm in process of fixing all the types. There's a lot of mix-n-match, sometimes separate types which are not exactly like in ol (like for some of conditions), some are straight wrong. Changing everything to "proxy" the types from ol options. Since I'm doing that on my off time, it's not going as fast. Depends on when the PR gets merged, either pushing to that or a new PR.

**Comment by azygis** [comment]
All typings have been corrected, using the exported `Options` types as close as vue compiler can handle. Intellisense works better as well now, including majority of events.

@d-koppenhagen @MelihAltintas can someone review? I'd like to have this merged first before starting to tackle #378.

**Comment by d-koppenhagen** [comment]
@azygis great job!, thanks for your contribution! I will cut a new version soon
